apps/sms/views/new_message/style/recipients.css

```diff
@@ -143,56 +143,25 @@ Do not remove.
  * to intelligently collapse when the final row contains no "assimilated"
  * (i.e. non-contenteditable) recipients. */
 .recipient[contenteditable=true]:empty {
-  /**
-   * We don't use -moz-margin-end here because of the bug 1103011, as currently
-   * directionality of empty contenteditable isn't inherited from parent, but
-   * always LTR, so -moz-margin-end equals to margin-right in both LTR and RTL
-   * modes. So to align it correctly we should override margin depending on text
-   * direction of parent element manually.
-   */
-  margin-right: -0.5rem;
   padding: 0;
 }
-
-.recipient[contenteditable=true]:focus,
-.recipient[contenteditable=true]:hover,
-.recipient[contenteditable=true]:empty:focus {
-  -moz-margin-end: 0.5rem;
-  -moz-padding-end: 0.3rem;
+html[dir="ltr"] .recipient[contenteditable=true]:empty {
+  margin-right: -0.5rem;
 }
-
-/* RTL for recipient panel */
-:-moz-dir(rtl) > .recipient[contenteditable] {
-  margin-left: 0.6rem;
-  margin-right: 0;
-
-  /**
-   * We need "text-align: right" only for the cases when currently editable
-   * recipient looses focus but not yet assimilated: case №1 - when we tap on
-   * thread-container - maybe fixed soon, case №2 - when we use screen reader,
-   * eg. single touch on message input steals focus from currently editable
-   * recipient, but doesn't yet give it to the message input. As the result
-   * flex-grow (see workaround for bug 1103011) returns and because of dir=auto
-   * LTR\RTL name would be always left\right aligned in both LTR and RTL modes.
-   */
-  text-align: right;
+html[dir="rtl"] .recipient[contenteditable=true]:empty {
+  margin-left: -0.5rem;
 }
 
-:-moz-dir(rtl) > .recipient[contenteditable].email {
-  /**
-   * See LTR ".recipient[contenteditable].email" rule above for the explanation
-   * why we don't use -moz-padding-start/-end here.
-   */
-  padding: 0 3.8rem 0 1.5rem;
-
-  background-position: right 1.5rem center;
+html[dir="ltr"] .recipient[contenteditable=true]:focus,
+html[dir="ltr"] .recipient[contenteditable=true]:hover,
+html[dir="ltr"] .recipient[contenteditable=true]:empty:focus {
+  margin-right: 0.5rem;
+  padding-right: 0.3rem;
 }
-
-:-moz-dir(rtl) > .recipient[contenteditable=true]:empty {
-  /**
-   * See LTR ".recipient[contenteditable=true]:empty" rule above for the
-   * explanation why we don't use -moz-margin-start/-end here.
-   */
-  margin-left: -0.5rem;
-  margin-right: 0;
+html[dir="rtl"] .recipient[contenteditable=true]:focus,
+html[dir="rtl"] .recipient[contenteditable=true]:hover,
+html[dir="rtl"] .recipient[contenteditable=true]:empty:focus {
+  margin-left: 0.5rem;
+  padding-left: 0.3rem;
 }
+
```

Comment:
you did a really good job here, this is a very tricky part of the SMS app and I can't find any issue !

Still i'd like that you try to keep as many comments as you can, because they give useful explanations about some choices we made here.